scipy/optimize/lbfgsb.py

```diff
@@ -275,27 +286,51 @@ def _minimize_lbfgsb(fun, x0, args=(), jac=None, bounds=None,
         bounds = [(None, None)] * n
     if len(bounds) != n:
         raise ValueError('length of x0 != length of bounds')
+
     # unbounded variables must use None, not +-inf, for optimizer to work properly
     bounds = [(None if l == -np.inf else l, None if u == np.inf else u) for l, u in bounds]
+    # LBFGSB is sent 'old-style' bounds, 'new-style' bounds are required by
+    # approx_derivative and ScalarFunction
+    new_bounds = old_bound_to_new(bounds)
+    # initial vector must lie within the bounds. Otherwise ScalarFunction and
+    # approx_derivative will cause problems
+    x0 = np.clip(x0, new_bounds[0], new_bounds[1])
 
     if disp is not None:
         if disp == 0:
             iprint = -1
         else:
             iprint = disp
 
-    n_function_evals, fun = wrap_function(fun, ())
+    n_function_evals, func = wrap_function(fun, args)
+
     if jac is None:
         def func_and_grad(x):
-            f = fun(x, *args)
-            g = _approx_fprime_helper(x, fun, epsilon, args=args, f0=f)
+            f = func(x)
+            g = approx_derivative(func, x, abs_step=eps, f0=f,
+                                  method='2-point', bounds=new_bounds)
             return f, g
-    else:
+
+    elif callable(jac):
         def func_and_grad(x):
-            f = fun(x, *args)
-            g = jac(x, *args)
+            return func(x), jac(x, *args)
```

Comment:
`func(x,*args)`?

Reply:
`func` is [`fun` wrapped with `args`](https://github.com/scipy/scipy/pull/10673/files#diff-be9847790086f0afe52f687cc3ea9038R305)